app/javascript/mastodon/components/account_bio.tsx

```diff
@@ -15,6 +33,28 @@ export const AccountBio: React.FC<{
       className={`${className} translate`}
       dangerouslySetInnerHTML={{ __html: note }}
       onClickCapture={handleClick}
+      ref={handleNodeChange}
     />
   );
 };
+
+function addDropdownToHashtags(node: HTMLElement | null, accountId: string) {
+  if (!node) {
+    return;
+  }
+  for (const childNode of node.childNodes) {
+    if (!(childNode instanceof HTMLElement)) {
+      continue;
+    }
+    if (
+      childNode instanceof HTMLAnchorElement &&
+      (childNode.classList.contains('hashtag') ||
+        childNode.innerText.startsWith('#')) &&
+      !childNode.dataset.menuHashtag
+    ) {
+      childNode.dataset.menuHashtag = accountId;
+    } else if (childNode.childNodes.length > 0) {
+      addDropdownToHashtags(childNode, accountId);
+    }
+  }
+}
```

Comment:
@ChaosExAnima Would it be better to assign the data-menu-hashtag attribute at a lower-level, i.e. wherever the hashtag link formatting is done and the `rel="tag"` attribute is added? (Probably in Rubyland, no idea how easy it'd be…)

Reply:
yeah that's directly coming from Ruby. if you check out how it's being done in the `StatusContent` component it also adds the dropdown data this way. i don't feel this is good to put into Ruby as this dropdown menu implementation entirely based in React, so putting it into Ruby makes the implementation super messy.